### PR TITLE
Add timeout and max_retries options

### PR DESCRIPTION
**Pull request by sunny**
Add access to `timeout` and `max_retries` configuration options from Meilisearch’s HTTParty integration.

### CODE REVIEW

**Review by CaroFG — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -104,8 +104,12 @@ Create a new file `config/initializers/meilisearch.rb` to setup your `MEILISEARC
...
 ```ruby
 MeiliSearch.configuration = {
-    meilisearch_host: 'YourMeiliSearchHost',
-    meilisearch_api_key: 'YourMeiliSearchAPIKey',
+  meilisearch_host: 'YourMeiliSearchHost',
+  meilisearch_api_key: 'YourMeiliSearchAPIKey',
+
+  # Extra configuration:
+  # timeout: 2,
+  # max_retries: 1,
```
Comment (bidoubiwa) [inline README.md:112]:
```suggestion
```

Thanks for adding this in the README but we try to make it as small as possible to not confuse new users.  
The same way [`Backend Pagination`](https://github.com/meilisearch/meilisearch-rails/blob/3b4a57b7a3411456d1abd5007adeb353a425f51a/README.md#backend-pagination-) has its own section, Timeout and Max-retries could have their own section as well. 

```
### Extra Configuration

Requests made to MeiliSearch may timeout and retry. To adapt the behavior to your needs, you can change the parameters during configuration: 

MeiliSearch.configuration = {
  meilisearch_host: 'YourMeiliSearchHost',
  meilisearch_api_key: 'YourMeiliSearchAPIKey',

  timeout: 2,
  max_retries: 1,
}

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
Thanks so much for this addition 🔥

File: README.md (hunk 1 above)
Comment (sunny) [reply README.md]: 💯 

Done!

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

**Comment by bidoubiwa** [comment]
bors merge

**Comment by CaroFG** [comment]
bors merge

**Pull request merged**